x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/build_events_query.ts

```diff
@@ -74,7 +76,8 @@ export const buildEventsSearchQuery = ({
           ],
         },
       },
-      track_total_hits: true,
+      // if we have maxDocs, don't utilize track total hits.
+      track_total_hits: maxDocs != null ? false : true,
```

Comment:
See my comments below but I think that you are always going to have a maxDocs?
https://github.com/elastic/kibana/pull/50129/files#r345313497

I think its good that we always carry a maximum value and a default even if the user tries to get around it they will have to be very intentional in increasing the number to a very high number.

Let me know otherwise. We can always increase it to a very high default, but carrying a sensible lower default feels more like the same as the other API's from Elastic we work with.